delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resolvers/ResolverRejection.scala

```diff
@@ -0,0 +1,92 @@
+    * does not exist.
+    *
+    * @param provided the provided revision
+    * @param current  the last known revision
+    */
+  final case class RevisionNotFound(provided: Long, current: Long)
+      extends ResolverRejection(s"Revision requested '$provided' not found, last known revision is '$current'.")
+
+  /**
+    * Rejection returned when attempting to create a resolver with an id that already exists.
+    *
+    * @param id      the resolver identifier
+    * @param project the project it belongs to
+    */
+  final case class ResolverAlreadyExists(id: Iri, project: ProjectRef)
+      extends ResolverRejection(s"Resolver '$id' already exists in project '$project'.")
+
+  /**
+    * Rejection returned when attempting to update a resolver with an id that doesn't exist.
+    *
+    * @param id      the resolver identifier
+    * @param project the project it belongs to
+    */
+  final case class ResolverNotFound(id: Iri, project: ProjectRef)
+      extends ResolverRejection(s"Resolver '$id' not found in project $project.")
+
+  /**
+    * Rejection returned when attempting to create a resolver with an id that already exists.
+    *
+    * @param id the resolver identifier
+    */
+  final case class DifferentResolverType(id: Iri, found: String, expected: String)
+      extends ResolverRejection(s"Resolver '$id' is of type ''$found'' and can't be updated to be a ''$expected'' .")
+
+  /**
+    * Rejection returned when no identities has been provided
+    */
+  final case object NoIdentities extends ResolverRejection(s"At least one identity of the caller must be provided")
+
+  /**
+    * Rejection return when the logged caller does not have one of the provided identities
+    */
+  final case class InvalidIdentities(missingIdentities: Set[Identity])
+      extends ResolverRejection(
+        s"The caller doesn't have some of the provided identities: ${missingIdentities.mkString(",")}"
+      )
+
+  /**
+    * Rejection returned when a subject intends to perform an operation on the current resolver, but either provided an
+    * incorrect revision or a concurrent update won over this attempt.
+    *
+    * @param provided the provided revision
+    * @param expected the expected revision
+    */
+  final case class IncorrectRev(provided: Long, expected: Long)
+      extends ResolverRejection(
+        s"Incorrect revision '$provided' provided, expected '$expected', the resolver may have been updated since last seen."
+      )
+
+  /**
+    * Rejection returned when attempting to update/deprecate a resolver that is already deprecated.
+    *
+    * @param id the resolver identifier
+    */
+  final case class ResolverIsDeprecated(id: Iri) extends ResolverRejection(s"Resolver '$id' is deprecated.")
+
+  /**
+    * Rejection returned when the project or the organization the resolver belongs to has been deprecated
+    */
+  final case class ReadOnlyProject(project: ProjectRef)
```

Comment:
This is not the only error that can occur. It can also be that the project is not found.
However, why don't you just do:

```scala
  final case class WrappedProjectRejection(rejection: ProjectRejection) extends ResourceRejection(rejection.reason)
```

So that we don't end up having different client signaling messages for these things.